source-mysql/discovery.go

```diff
@@ -350,11 +363,18 @@ func (db *mysqlDatabase) translateRecordField(columnType interface{}, val interf
 				return t.UTC().Format(time.RFC3339Nano), nil
 			}
 		}
+		if len(val) > truncateColumnThreshold {
```

Comment:
under what conditions does this fire -- what does it mean for columnType to not be a `string`?

Reply:
The logic could definitely be made a bit more clear here, it just kind of grew like this over time. The `columnType` should always be a string at this point (because the only non-string types are handled up at the top of this function where we delegate to a separate `columnType.translateRecordField(val)` method for enums/sets/integers).

This return here is the default case for when the `switch typeName` doesn't have a special case for a particular column type.